intel-sgx/enclave-runner/src/usercalls/mod.rs

```diff
@@ -1214,6 +1235,33 @@ extern "C" fn handle_trap(_signo: c_int, _info: *mut siginfo_t, context: *mut c_
     return;
 }
 
+#[cfg(unix)]
+fn catch_sighup() {
+    unsafe {
+        extern "C" fn handle_sighup(_signo: c_int, _info: *mut siginfo_t, _context: *mut c_void) {
+            eprintln!("SIGHUP triggered, thread_id: {:?}", std::thread::current().id());
+            let instruction_ptr = unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] as *const u8};
+            // enclu instruction code
+            const ENCLU: [u8; 3] = [0x0f, 0x01, 0xd7];
+            let is_enclu = ENCLU.iter().enumerate().all(|(idx, v)| {
+                unsafe { *instruction_ptr.offset(idx as isize) == *v }
+            });
+            if is_enclu && EXITING.with(|cell| cell.borrow().as_ref().is_some_and(|val| val.load(Ordering::SeqCst) == true)) {
+                // Interrupt enclave execution by setting IP to the instruction following the ENCLU to mimic normal ENCLU[EXIT])
+                unsafe {
+                    (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] += 3;
```

Comment:
Can you also rename `_context` to `context`?

Reply:
done